pkg/dockerregistry/server/storagedriver.go

```diff
@@ -0,0 +1,32 @@
+package server
+
+import (
+	"regexp"
+
+	context "github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	registrystorage "github.com/docker/distribution/registry/storage/driver/middleware"
+)
+
+func init() {
+	registrystorage.Register("openshift", registrystorage.InitFunc(newLocalStorageDriver))
+}
+
+type localStorageDriver struct {
+	storagedriver.StorageDriver
+}
+
+var _ storagedriver.StorageDriver = &localStorageDriver{}
+
+var pathSplitRe = regexp.MustCompile("^(?P<repoPrefix>.*/repositories)/(?P<name>[^/]+/[^/]+)/(?P<content>.*)$")
+
+func newLocalStorageDriver(storageDriver storagedriver.StorageDriver, options map[string]interface{}) (storagedriver.StorageDriver, error) {
+	return &localStorageDriver{storageDriver}, nil
+}
+
+func (s *localStorageDriver) GetContent(ctx context.Context, path string) ([]byte, error) {
+	if reponame, ok := ctx.Value("openshift.storage.getcontent.reponame").(string); ok {
+		path = pathSplitRe.ReplaceAllString(path, "${repoPrefix}/"+regexp.QuoteMeta(reponame)+"/${content}")
```

Comment:
this is the part that concerns me... we're swapping part of a path to another reponame without verifying you have access to the specific content in that repo being requested, right?